offscreen-canvas/tools/templates.yaml

```diff
@@ -0,0 +1,58 @@
+w3c: |
+    <!DOCTYPE html>
+    <!-- DO NOT EDIT! This test has been generated by tools/gentest.py. -->
+    <title>OffscreenCanvas test: %(name)s</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="/common/canvas-tests.js"></script>
+
+    <h1>%(name)s</h1>
+    <p class="desc">%(desc)s</p>
+
+    %(notes)s
+    <script>
+    var t = async_test("%(escaped_desc)s");
+    t.step(function() {
+
+    var offscreenCanvas = new OffscreenCanvas(100, 50);
+    var ctx = offscreenCanvas.getContext('2d');
+
+    %(code)s
+    t.done();
+
+    });
+    </script>
+
+
+w3cmain: |
+    <!DOCTYPE html>
+    <!-- DO NOT EDIT! This test has been generated by tools/gentest.py. -->
+    <title>OffscreenCanvas test in a worker: %(name)s</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+
+    <h1>%(name)s</h1>
+    <p class="desc">%(desc)s</p>
+
+    %(notes)s
+    <script>
+    fetch_tests_from_worker(new Worker("%(name)s.js"));
+    </script>
+
+w3cworker: |
+    importScripts("/resources/testharness.js");
```

Comment:
The worker script should also have a comment stating that the file was generated and should not be edited directly.